src/game_enemy.cpp

```diff
@@ -169,6 +169,7 @@ int Game_Enemy::GetHue() const {
 
 void Game_Enemy::SetHidden(bool _hidden) {
 	hidden = _hidden;
+	ResetBattleStateOnDeath();
```

Comment:
I don't think SetHidden should change anything. They are just hidden and keep everything.
Because this is unconditionally enemies have 0 HP when the battle starts.

Also you forgot the edge case (bug?) that enemies removed via "Force Flee" in 2k3 can reappear with "Enemy encounter", another reason against "SetHp(0)" ;)